src/pudl_rmi/validate.py

```diff
@@ -0,0 +1,178 @@
+    "other_fossil": "Other",
+    "hydro": "Hydraulic",
+    "regional_transmission_and_market_operation": "Transmission",
+}
+
+
+def download_and_clean_net_plant_balance(pudl_engine: sa.engine.Engine) -> pd.DataFrame:
+    """
+    Download the Utility Transition Hub's cleaned FERC1 Net Plant Balance Table.
+
+    This function grabs the RMI Hub Team's cleaned version of the FERC1 Net
+    Plant Balance table. This function converts column names to PUDL column
+    names, standardizes FERC Account Names and merges in the ``utility_id_pudl``
+    through the ``utilities_ferc1`` PUDL table.
+
+    Note: In the fullness of time, this table should really come directly from
+    PUDL or the FERC1 database.
+
+    Args:
+        pudl_engine: A connection engine for the PUDL DB.
+    """
+    utils_f1 = pd.read_sql("utilities_ferc1", pudl_engine)
+    npb = pd.read_csv(
+        "https://utilitytransitionhub.rmi.org/static/data_download/net_plant_balance.csv"
+    )
+    npb = (
+        npb.convert_dtypes(convert_floating=False)
+        .rename(columns=RMI_TO_PUDL_COLS)
+        .replace({"ferc_acct_name": RMI_TO_PUDL_FERC_ACCT_NAME})
+        .assign(report_date=lambda x: pd.to_datetime(x.report_year, format="%Y"))
+        .merge(  # we need the pudl id!!
+            utils_f1[["utility_id_ferc1", "utility_id_pudl"]],
+            on=["utility_id_ferc1"],
+            how="left",
+            validate="m:1",
+        )
+    )
+    npb.loc[:, "ferc_acct_name"] = npb.ferc_acct_name.str.title()
+    return npb
+
+
+def add_data_source(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Add a data_source column with 'FERC' and 'PUC'.
+
+    The depreciation data includes a `data_source` column as a part of the
+    primary key of that data set. This is because there are sometimes we have
+    depreciation study from both `FERC` and `PUC` for the same utility and
+    plants. Because of this, in order to compare the FERC plant input tables to
+    the FERC-EIA-Depreciation tables we need to add *both* versions of the
+    depreciation `data_source`.
+    """
+    if "data_source" not in df:
+        df = pd.concat(
+            [
+                df.assign(data_source="PUC"),
+                df.assign(data_source="FERC"),
+            ]
+        )
+    return df
+
+
+def group_sum_cols(df, data_cols: List[str], by: List[str]) -> pd.DataFrame:
+    """Groupby sum a specific table's data cols."""
+    # convert date to year bc many of the og depish studies are EOY
+    summed_out = (
+        df.assign(report_year=lambda x: x.report_date.dt.year)
+        .astype({"report_year": pd.Int64Dtype()})
+        .groupby(by=by, dropna=True)[data_cols]
+        .sum(min_count=1)
+    )
+    return summed_out
+
+
+def agg_test_data(
+    df1: pd.DataFrame, df2: pd.DataFrame, data_cols: List[str], by: List[str], **kwarg
+) -> pd.DataFrame:
+    """
+    Merge two grouped input tables to determine if summed data column are equal.
+
+    Args:
```

Comment:
Include `**kwargs` as an argument and tell the user what function the `kwargs` are going to get passed on to so they can use the function correctly without needing to look at the source.

Reply:
Was `**kwarg` supposed to become singular instead of `**kwargs` (which I think is the standard name)?

It's not clear what "a ratio column is the divsion of the sum of the data column from df1 and df2." means.

I think you mean:
> The ratio of the aggregated data columns in df1 and df1 (df1 / df2)